README.md
# Crawl Splash

A flutter package that helps you build beautiful splash screens for your app.

## Why We Build?

Can you build a **Splash screen within minutes** for your app? This is where we help you do that.

![Build Status](https://img.shields.io/badge/Build-Passing-blue) ![Build Status](https://img.shields.io/badge/Release-V0.1.0-green) ![Build Status](https://img.shields.io/badge/license-MIT-red) ![Build Status](https://img.shields.io/badge/Contribution-Open-Yellow) ![Build Status](https://img.shields.io/badge/Platform-Flutter-orange) ![Build Status](https://img.shields.io/badge/Language-Dart-DarkPink)

## Screenshots

<img src="https://raw.githubusercontent.com/imsks/Crawl-Splash/master/screenshots/crawl-splash-img-1.jpg"  height = "400" alt="PieChart">

## Usage

### Import this class

```dart
import 'package:crawlsplash/src/basic_crawl_splash_screen.dart';
```

### Add the On-Boarding screen

```dart
import 'package:flutter/material.dart';
import 'package:crawlsplash/src/basic_crawl_splash_screen.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      // Give Splash Screen Path as First Screen
      initialRoute: BasicSplashScreen.id,
      routes: {
        BasicSplashScreen.id: (context) => BasicSplashScreen(
              // Background color of the screen
              backgroundColor: const [
                Colors.white,
              ],

              // Time for which screen will show up
              duration: 1,

              // Image path => Asset Image ONLY
              imagePath: 'assets/images/instagram.png',

              // Image Height
              iconHeight: 100,

              // Image Width
              iconWidth: 200,

              // Splash screen text => Not Required
              primaryText: 'Instagram',

              // Screen Text Color
              primaryTextColor: Colors.pink,

              // Screen Text Font
              primaryTextFont: 'Roboto',

              // Screen Text Size
              primaryTextSize: 30,

              // Screen Text Font Weight
              primaryTextWeight: FontWeight.w400,

              // Replace null with your landing screen route
              nextScreenPath: null,

              // Change it to true to check how your splash screen looks like
              debug: false,
            ),
      },
    );
  }
}

```

## Features

- You can customize your Splash Screens with content and styling.

## Contribution

**The project is open for contribution. Anyone willing to code fresh splash screens, customize content for users will be welcomed.**

1. Fork it
2. Create your feature branch (git checkout -b my-new-feature)
3. Commit your changes (git commit -m 'Add some feature')
4. Push to the branch (git push origin my-new-feature)
5. Create new Pull Request


CHANGELOG.md
# Changelog

## [0.0.1] - 2020-09-21
- Launching V0.1.0

lib/src/basic_crawl_splash_screen.dart
import 'package:flutter/material.dart';
import 'dart:async';
import 'package:crawl_splash/utilities/bgDecoration.dart';

class BasicSplashScreen extends StatefulWidget {
  static const String id = 'splashScreen';

  // Background color of the screen
  @required
  final List<Color> backgroundColor;

  // Time for which screen will show up
  @required
  final int duration;

  // Image path => Asset Image ONLY
  @required
  final String imagePath;

  // Image Height
  final double iconHeight;

  // Image Width
  final double iconWidth;

  // Next screen it will land after the splash screen => Route name
  @required
  final String nextScreenPath;

  // Splash screen text => Not Required
  final String primaryText;

  // Screen Text color
  final Color primaryTextColor;

  // Screen Text Size
  final double primaryTextSize;

  // Screen Text font
  final String primaryTextFont;

  // Screen Text font
  final FontWeight primaryTextWeight;

  // Check if user is debugging
  final bool debug;

  BasicSplashScreen(
      {this.backgroundColor,
      this.duration,
      this.imagePath,
      this.nextScreenPath,
      this.primaryText,
      this.iconHeight,
      this.iconWidth,
      this.primaryTextColor,
      this.primaryTextFont,
      this.primaryTextSize,
      this.primaryTextWeight,
      this.debug});

  @override
  _BasicSplashScreenState createState() => _BasicSplashScreenState();
}

class _BasicSplashScreenState extends State<BasicSplashScreen> {
  @override
  void initState() {
    super.initState();
    if (!widget.debug && widget.nextScreenPath != "") {
      new Future.delayed(
        Duration(seconds: widget.duration),
        () => Navigator.pushReplacementNamed(context, widget.nextScreenPath),
      );
    }

    if (widget.nextScreenPath != "") {
      print(
          "You've not given Landing Screen Path. Please give the path as string as explained in doc.");
    }
  }

  Widget build(BuildContext context) {
    return Container(
      width: double.infinity,
      height: double.infinity,
      decoration: bgBoxDecoration(widget.backgroundColor),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: <Widget>[
          Image.asset(
            widget.imagePath,
            height: widget.iconHeight,
            width: widget.iconWidth,
          ),
          Text(
            widget.primaryText,
            style: TextStyle(
              decoration: TextDecoration.none,
              color: widget.primaryTextColor,
              fontFamily: widget.primaryTextFont,
              fontSize: widget.primaryTextSize,
              fontWeight: widget.primaryTextWeight,
            ),
          ),
        ],
      ),
    );
  }
}


example/lib/example.dart
import 'package:flutter/material.dart';
import 'package:crawl_splash/crawlsplash.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      // Give Splash Screen Path as First Screen
      initialRoute: BasicSplashScreen.id,
      routes: {
        BasicSplashScreen.id: (context) => BasicSplashScreen(
              // Background color of the screen
              backgroundColor: const [
                Colors.white,
              ],

              // Time for which screen will show up
              duration: 1,

              // Image path => Asset Image ONLY
              imagePath: 'assets/images/instagram.png',

              // Image Height
              iconHeight: 100,

              // Image Width
              iconWidth: 200,

              // Splash screen text => Not Required
              primaryText: 'Instagram',

              // Screen Text Color
              primaryTextColor: Colors.pink,

              // Screen Text Font
              primaryTextFont: 'Roboto',

              // Screen Text Size
              primaryTextSize: 30,

              // Screen Text Font Weight
              primaryTextWeight: FontWeight.w400,

              // Replace null with your landing screen route
              nextScreenPath: null,

              // Change it to true to check how your splash screen looks like
              debug: false,
            ),
      },
    );
  }
}
